pipeline_utils/pipeline_deploy.py

```diff
@@ -0,0 +1,439 @@
+                            write_.write(line)
+                # upload to s3
+                extra_args = {'ACL': 'public-read'}  # note that this is no longer public if using encryption!
+                if self.kms_key_id:
+                    extra_args.update(update_)
+                s3.meta.client.upload_file(upload_file_, self.wfl_bucket, s3_file_, ExtraArgs=extra_args)
+                logger.info('> Posted %s' % s3_file_)
+                # delete file to allow tmp folder to be deleted at the end
+                os.remove(upload_file_)
+
+        # Clean tmp directory
+        os.rmdir(upload_)
+
+    def _post_patch_ecr(self, type='ECR'):
+        """
+        """
+        logger.info('@ Docker Image...')
+
+        # Set general variables
+        filepath_ = f'{self.repo}/{self.filepath[type]}'
+        account_ = f'{self.account}.dkr.ecr.{self.region}.amazonaws.com'
+
+        # Check
+        if not os.path.isdir(filepath_):
+            logger.error(f'WARNING: {self.filepath[type]} not found in {self.repo}, skipping...')
+            return
+
+        # Create ecr object
+        ecr = boto3.client('ecr')
+        response = ecr.describe_repositories()
+
+        # Generic bash commands to be modified to correct version and account information
+        for fn in map(os.path.basename, glob.glob(f'{filepath_}/*')):
+            logger.info('> Processing %s' % fn)
+            if not self.debug:
+                # set specific variables
+                tag_ = f'{account_}/{fn}:{self.version}'
+                path_ = f'{filepath_}/{fn}'
+                is_repository = False
+                # check if tag is present in ECR repositories,
+                #   if not create it
+                for repository in response['repositories']:
+                    if repository['repositoryArn'].split('/')[-1] == fn:
+                        is_repository = True
+                        break
+                if not is_repository:
+                    logger.info('> Creating ECR Repository %s' % fn)
+                    ecr.create_repository(repositoryName=fn)
+                # build and push the image
+                # do so by triggering a CodeBuild run
+                build_projects = self._codebuild.list_projects()
+                builder = list(filter(lambda b: f'{self.ff_env}-pipeline-builder' == b, build_projects))
+                if not builder:
+                    logger.info('NOTE: no builder job found! Triggering a local build.'
+                                ' Ensure CodeBuild jobs are present by deploying the codebuild stack')
+                    image = f"""
+                                                aws ecr get-login-password --region {self.region} | docker login --username AWS --password-stdin {account_}
+                                                docker buildx -t {tag_} {path_} --no-cache --platform linux/amd64,linux/arm64
+                                                docker push {tag_}
```

Comment:
Wouldn't be worth removing images after we push them to ECR by running: `docker image rm {tag_}` to save space on the disk? I was deploying all the new YAML based pipelines and at some point run out of memory.
That would remove the image only but layers would stay  (for example: our base docker `cgap/cgap-ubuntu2004-py-38 `)  but didn't find an automated way to remove them as well.

Reply:
Definitely a good idea, but in the newest release this won't be an issue because the code I added will trigger the builds all through CodeBuild, so there will be no images built/pushed locally anymore

Reply:
awesome! What are the other differences of using CodeBuild?

Reply:
removed local build for now and Will added codebuild